### PR TITLE
chore(core): split strapi/strapi & strapi/core

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Split `strapi/strapi` into 2 packages:
- `strapi/core` contains all the `strapi/strapi` codebase except the CLI
- `strapi/strapi` keeps the CLI & re-exports everything  from `strapi/core` to keep backward compatibility
- This does not solve circular deps yet.

### Why is it needed?

Better pkg org and removing circular dep eventually

### How to test it?

- This PR should have no visible effect on anything. It's purely code movements.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

**Comment by joshuaellis** [comment]
Sorry, I find the PR description a bit unclear, can you explain what exactly this specific PR does please? I just want to make sure I QA / review correctly 😄

**Comment by alexandrebodin** [comment]
> Sorry, I find the PR description a bit unclear, can you explain what exactly this specific PR does please? I just want to make sure I QA / review correctly 😄

Sure, done. I'll remove you as a reviewer as it's not moving the admin commands yet and the rest is BE

File: packages/cli/create-strapi-app/package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
   "main": "",
   "bin": "./bin/index.js",
   "files": [
-    "./dist",
-    "./bin"
+    "dist/",
```
Comment (Convly) [inline packages/cli/create-strapi-app/package.json:36]: Are those purely codestyle improvements or does it serve a purpose?

File: packages/core/strapi/src/index.ts [hunk 2]
```diff
@@ -7,9 +7,6 @@ export type * from '@strapi/types';
 
 export type { StrapiCommand, CLIContext } from './commands/types';
 
-export * as factories from './factories';
+// export * as factories from './factories';
```
Comment (Convly) [inline packages/core/strapi/src/index.ts:10]: Why has this been removed?

File: packages/cli/create-strapi-app/package.json (hunk 1 above)
Comment (Convly) [reply packages/cli/create-strapi-app/package.json:36]: (same question for everywhere this change is present)

File: package.json [hunk 3]
```diff
@@ -142,6 +142,7 @@
     "supertest": "6.3.3",
     "ts-jest": "29.1.0",
     "typescript": "5.3.2",
+    "yalc": "1.0.0-pre.53",
```
Comment (Convly) [inline package.json:145]: Why are we using a pre-releasing?

File: packages/core/core/.eslintignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+node_modules/
+.eslintrc.js
+jest.config.js
+dist/
+index.d.ts
+index.js
+bin/
```
Comment (Convly) [inline packages/core/core/.eslintignore:7]: We don't have binaries in core/core, do we?

File: package.json (hunk 3 above)
Comment (alexandrebodin) [reply package.json:145]: because they are not in 1.0.0 yet it's the latest

File: packages/core/strapi/src/index.ts (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/strapi/src/index.ts:10]: I'll remove it for now because I'm going to change the API

**Review by Convly — APPROVED** [review]

**Pull request merged**